exchanges/exchange.go

```diff
@@ -340,6 +341,24 @@ func (e *Base) GetEnabledPairs(assetType asset.Item) currency.Pairs {
 	return pairs.Format(format.Delimiter, format.Index, format.Uppercase)
 }
 
+// GetRequestFormattedPairAndAssetType is a method that returns the enabled currency pair of
+// along with its asset type
+func (e *Base) GetRequestFormattedPairAndAssetType(p string) (currency.Pair, asset.Item, error) {
```

Comment:
Only issue with this and the one above is if we have a duplicate in assets i.e. BTC-USD for spot, index like in OKEX and it might come back with a false positive.

Reply:
I've added additional commentary to suggest not using it when there are duplicates. OKEX has not implemented this. One which did have an issue (coinbene) only has SWAP support for websocket right now, so I've used `NewPairFromFormattedPairs` there

Reply:
Awesome, thanks!